### PR TITLE
new tasks: cfn_update and get_kms_data_key()

### PR DESCRIPTION
**Pull request by ltsampros**

### CODE REVIEW

File: bootstrap_salt/fab_tasks.py [hunk 1]
```diff
@@ -145,7 +145,41 @@ def create_kms_data_key():
...
+        ips = None
+
+    #
+    # If we don't currently have any instances just create a new key
+    #
+    # XXX: Other possible scenario is to fetch it from within the
+    # LaunchConfiguration
+    #
+    if not ips:
+        return create_kms_data_key()
```
Comment (ltsampros) [inline bootstrap_salt/fab_tasks.py]: Could someone comment on this/review? My hypothesis here is that if get_instance_ips() doesn't return any IPs, we generate a new data_key anyway.

**Comment by filipposc5** [comment]
I was testing this and it is not working for me.
```
Traceback (most recent call last):
  File "/Users/filippos/.virtualenvs/mps-deploy/lib/python2.7/site-packages/fabric/main.py", line 756, in main
    *args, **kwargs
  File "/Users/filippos/.virtualenvs/mps-deploy/lib/python2.7/site-packages/fabric/tasks.py", line 426, in execute
    results['<local-only>'] = task.run(*args, **new_kwargs)
  File "/Users/filippos/.virtualenvs/mps-deploy/lib/python2.7/site-packages/fabric/tasks.py", line 173, in run
    return self.wrapped(*args, **kwargs)
  File "/Users/filippos/.virtualenvs/mps-deploy/lib/python2.7/site-packages/bootstrap_cfn/fab_tasks.py", line 701, in cfn_update
    new_body = cfn_config.process_update(body)
  File "/Users/filippos/.virtualenvs/mps-deploy/lib/python2.7/site-packages/bootstrap_cfn/config.py", line 132, in process_update
    ec2 = self.ec2()
  File "/Users/filippos/.virtualenvs/mps-deploy/lib/python2.7/site-packages/bootstrap_cfn/config.py", line 1092, in ec2
    user_data = self.get_ec2_userdata()
  File "/Users/filippos/.virtualenvs/mps-deploy/lib/python2.7/site-packages/bootstrap_salt/config.py", line 83, in get_ec2_userdata
    'content': self.kms_data_key,
AttributeError: 'MyConfigParser' object has no attribute 'kms_data_key'
```

**Comment by yufangzhang** [comment]
I had the same error as @filipposc5 ^
I think for some reason, it didn't start from bootstrap_salt.cfn_update therefore `env.kms_data_key` was empty. Because when I run cfn_update, it didn't pause at the breakpoint below, but went straight to bootstrap_cfn instead:
```
# bootstrap_salt/fab_task.py
@task
@wraps(bcfn_update)
def cfn_update(*args, **kwargs):
    import pdb;pdb.set_trace()
    env.kms_key_id = get_kms_key_id()
    env.kms_data_key = get_kms_data_key()
    bcfn_update(*args, **kwargs)
```

**Comment by yufangzhang** [comment]
When I resized EC2 instance, it created a new instance with that new InstanceType, but it was not attached to the existing ELB and ASG. Also current changes made docker upgrade by default, it might break old stacks?

**Comment by ltsampros** [comment]
@yufangzhang Do you have any logs to provide? Do you have the diff output?

File: requirements.txt [hunk 2]
```diff
@@ -5,7 +5,7 @@ mock==1.0.1
 dnspython
 python-slugify>=1.2.4
 testfixtures>=4.1.2
-git+https://github.com/ministryofjustice/bootstrap-cfn.git@cfn_update-ec2
+git+https://github.com/ministryofjustice/bootstrap-cfn.git
```
Comment (ltsampros) [inline requirements.txt]: that should be bootstrap-cfn>=1.4.3

**Review by yufangzhang — APPROVED** [review]

**Pull request merged**